### PR TITLE
protobuf: make toStatusProto return null when metadata key is absent

### PR DESCRIPTION
**Pull request by ericgribkoff**

### CODE REVIEW

File: protobuf/src/main/java/io/grpc/protobuf/StatusProto.java [hunk 1]
```diff
@@ -158,7 +158,7 @@ private static Metadata toMetadata(com.google.rpc.Status statusProto, Metadata m
 
   @Nullable
   private static com.google.rpc.Status toStatusProto(Status status, Metadata trailers) {
-    if (trailers != null) {
+    if (trailers != null && trailers.containsKey(STATUS_DETAILS_KEY)) {
       com.google.rpc.Status statusProto = trailers.get(STATUS_DETAILS_KEY);
       checkArgument(
```
Comment (carl-mastrangelo) [inline protobuf/src/main/java/io/grpc/protobuf/StatusProto.java]: The check for null should go here.  metadata uses a lot of linear time accesses, so doing it twice is reaching.

Comment (ericgribkoff) [inline protobuf/src/main/java/io/grpc/protobuf/StatusProto.java]: Done, PTAL.

**Comment by ericgribkoff** [comment]
Friendly ping...this fixes a null pointer exception, so would like to get this in before the branch cut.

File: protobuf/src/main/java/io/grpc/protobuf/StatusProto.java [hunk 2]
```diff
@@ -160,10 +160,12 @@ private static Metadata toMetadata(com.google.rpc.Status statusProto, Metadata m
   private static com.google.rpc.Status toStatusProto(Status status, Metadata trailers) {
     if (trailers != null) {
       com.google.rpc.Status statusProto = trailers.get(STATUS_DETAILS_KEY);
-      checkArgument(
-          status.getCode().value() == statusProto.getCode(),
-          "com.google.rpc.Status code must match gRPC status code");
-      return statusProto;
+      if (statusProto != null) {
+        checkArgument(
+                status.getCode().value() == statusProto.getCode(),
```
Comment (carl-mastrangelo) [inline protobuf/src/main/java/io/grpc/protobuf/StatusProto.java]: indent -4.

File: protobuf/src/main/java/io/grpc/protobuf/StatusProto.java [hunk 3]
```diff
@@ -160,10 +160,12 @@ private static Metadata toMetadata(com.google.rpc.Status statusProto, Metadata m
   private static com.google.rpc.Status toStatusProto(Status status, Metadata trailers) {
     if (trailers != null) {
```
Comment (carl-mastrangelo) [inline protobuf/src/main/java/io/grpc/protobuf/StatusProto.java]: Does there need to be a `checkNotNull(status, "status")` here?

Comment (ericgribkoff) [inline protobuf/src/main/java/io/grpc/protobuf/StatusProto.java]: No, it's not a nullable field of StatusException or StatusRuntimeException.

File: protobuf/src/main/java/io/grpc/protobuf/StatusProto.java (hunk 2 above)
Comment (ericgribkoff) [inline protobuf/src/main/java/io/grpc/protobuf/StatusProto.java]: I can do this when I'm at a computer later tonight. I think the +8 was the result of running google format, and allowed by the style guide when there is more than two continuation lines.

Comment (ericgribkoff) [inline protobuf/src/main/java/io/grpc/protobuf/StatusProto.java]: Indentation fixed.

**Pull request merged**